src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1093,11 +1093,23 @@ private static String federatedFlowStep4(
       // step 5
       String postBackUrl = getPostBackUrlFromHTML(responseHtml);
       if (!isPrefixEqual(postBackUrl, loginInput.getServerUrl())) {
-        logger.debug(
-            "The specified authenticator {} and the destination URL "
-                + "in the SAML assertion {} do not match.",
-            loginInput.getAuthenticator(),
-            postBackUrl);
+        URL idpDestinationUrl = new URL(postBackUrl);
+        URL clientDestinationUrl = new URL(loginInput.getServerUrl());
+        String idpDestinationHostName = idpDestinationUrl.getHost();
+        String clientDestinationHostName = clientDestinationUrl.getHost();
+
+        logger.error(
+            "The Snowflake hostname specified in the client connection {} does not match "
+                + "the destination hostname in the SAML response returned by the IdP: {}",
+            clientDestinationHostName,
+            idpDestinationHostName);
+
+        System.err.println(
```

Comment:
If it's for testing then it's fine. Please remove it before the merge.

Reply:
I'm printing that out to the stderr stream because otherwise, nobody is going to understand what the error is until they go and explicitly enable logging for the JDBC driver. It's a significant error too.

Reply:
I removed it for now until I can think of a better way to get messages into stderr in a cleaner way in the future.